### PR TITLE
Migration of order view page printing functionality and layout changes - part 3

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Rebuild of PR https://github.com/PrestaShop/PrestaShop/pull/16700 . When clicked "print" page display its version ready for print. There are some layout changes when print is being initialized.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #15819 
| How to test?  | Try clicking print or ctrl + p while in this page **admin-dev/index.php/sell/orders/orders/1/view**. It should be similar to invision sketch of when page is printed. **Dont forget to rebuild assets**

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17617)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig [hunk 1]
```diff
@@ -23,46 +23,90 @@
...
+        {{ merchantReturnsTitle }}
       </a>
     </li>
   </ul>
 
   <div class="tab-content">
-    <div class="tab-pane fade show active" id="historyTabContent" role="tabpanel" aria-labelledby="historyTab">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/history.html.twig' %}
+    <div class="tab-pane d-print-block fade show active" id="historyTabContent" role="tabpanel" aria-labelledby="historyTab">
+      {% embed '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig' %}
```
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:72]: @NeOMakinG legit ?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig [hunk 2]
```diff
@@ -23,46 +23,90 @@
...
+        {% endblock %}
+        {% block body %}
+          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/history.html.twig' %}
+        {% endblock %}
+      {% endembed %}
     </div>
-    <div class="tab-pane fade" id="orderDocumentsTabContent" role="tabpanel" aria-labelledby="orderDocumentsTab">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/documents.html.twig' %}
+    <div class="tab-pane d-print-block fade" id="orderDocumentsTabContent" role="tabpanel" aria-labelledby="orderDocumentsTab">
+      {% embed '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig' %}
```
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:82]: @NeOMakinG legit ? it's the same as before, duplicate 😅 ?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig [hunk 3]
```diff
@@ -23,46 +23,90 @@
...
+        {% endblock %}
+        {% block body %}
+          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/documents.html.twig' %}
+        {% endblock %}
+      {% endembed %}
     </div>
-    <div class="tab-pane fade" id="orderShippingTabContent" role="tabpanel" aria-labelledby="orderShippingTab">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/shipping.html.twig' %}
+    <div class="tab-pane d-print-block fade" id="orderShippingTabContent" role="tabpanel" aria-labelledby="orderShippingTab">
+      {% embed '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig' %}
```
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:92]: Again 😄

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig [hunk 4]
```diff
@@ -23,46 +23,90 @@
...
+        {% endblock %}
+      {% endembed %}
     </div>
-    <div class="tab-pane fade" id="orderReturnsTabContent" role="tabpanel" aria-labelledby="orderReturnsTab">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/merchandise_returns.html.twig' %}
+    <div class="tab-pane d-print-block fade" id="orderReturnsTabContent" role="tabpanel" aria-labelledby="orderReturnsTab">
+      {% embed '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig' %}
+        {% block header %}
+          {{ merchantReturnsTitle }}
+        {% endblock %}
```
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:105]: And again

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig (hunk 1 above)
Comment (NeOMakinG) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:72]: It's a really good practice to be honest, avoiding duplicate code and too much files for the same feature, it's fine for me

File: admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js [hunk 5]
```diff
@@ -24,153 +24,166 @@
...
+  orderPaymentFormAmountInput: "#order_payment_amount",
+  orderPaymentInvoiceSelect: "#order_payment_id_invoice",
+  viewOrderPaymentsBlock: "#view_order_payments_block",
+  privateNoteToggleBtn: ".js-private-note-toggle-btn",
+  privateNoteBlock: ".js-private-note-block",
+  privateNoteInput: "#private_note_note",
+  privateNoteSubmitBtn: ".js-private-note-btn",
+  addCartRuleModal: "#addOrderDiscountModal",
+  addCartRuleApplyOnAllInvoicesCheckbox:
+    "#add_order_cart_rule_apply_on_all_invoices",
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js]: Why do we have these weird line breaks?

File: admin-dev/themes/new-theme/scss/pages/_orders_view.scss [hunk 6]
```diff
@@ -0,0 +1,72 @@
+#order-view-page {
+  @media not print {
+    .card--details {
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/scss/pages/_orders_view.scss]: Double dash?

File: admin-dev/themes/new-theme/scss/components/_button.scss [hunk 7]
```diff
@@ -1,12 +1,12 @@
 .btn {
   &.btn-action {
-    color: #576C72;
-    background-color: white;
+    color: #000;
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/scss/components/_button.scss]: @NeOMakinG Color change is wanted  ?

File: admin-dev/themes/new-theme/template/components/layout/nav_bar.tpl [hunk 8]
```diff
@@ -22,7 +22,7 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  * International Registered Trademark & Property of PrestaShop SA
  *}
-<nav class="nav-bar d-none d-md-block">
+<nav class="nav-bar d-none d-print-none d-md-block">
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/template/components/layout/nav_bar.tpl:25]: Shouldn't this be handled using css selectors?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<div class="card card--details">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig]:
```suggestion
<div class="card card-details">
```

File: admin-dev/themes/new-theme/scss/pages/_orders_view.scss (hunk 6 above)
Comment (Progi1984) [inline admin-dev/themes/new-theme/scss/pages/_orders_view.scss]:
```suggestion
    .card-details {
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig [hunk 10]
```diff
@@ -196,14 +196,14 @@
     </div>
     <div class="mt-2 info-block">
       <div class="row">
+        {% set isPrivateNoteOpen = not orderForViewing.customer.privateNote is empty %}
+
         <div class="col-md-6">
-          <h3 class="mb-0">
+          <h3 class="mb-0 {{ not isPrivateNoteOpen ? 'd-print-none': '' }}">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig]:
```suggestion
          <h3 class="mb-0{{ not isPrivateNoteOpen ? ' d-print-none': '' }}">
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig [hunk 11]
```diff
@@ -61,6 +61,15 @@
     </form>
   {% endif %}
 
+  <form class="form-inline d-inline-block form-horizontal ml-2">
+    <div class="input-group">
+      <button type="button" class="btn btn-action js-print-order-view-page">
+        <i class="material-icons">print</i>
```
Comment (eternoendless) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig]: Missing aria attribute (this should be done for ALL decorative icons)
```suggestion
        <i class="material-icons" aria-hidden="true">print</i>
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_order_statistics.html.twig [hunk 12]
```diff
@@ -0,0 +1,42 @@
+<div class="card">
+  <div class="card-body">
+    <div class="row">
+      <div class="col-sm">
+        <p>
+          <strong>{{ 'Date'|trans({}, 'Admin.Global') }}</strong>
+        </p>
+        <strong>{{ orderForViewing.createdAt|date('Y-m-d') }}</strong>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_order_statistics.html.twig]: Shouldn't be localized to the user ?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_title.html.twig [hunk 13]
```diff
@@ -0,0 +1,7 @@
+<span class="h1 print-title">
+  {{ 'Order'|trans({}, 'Admin.Global') }}
+  <span class="print-title__id">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_title.html.twig]:
```suggestion
  <span class="print-title-id">
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig [hunk 14]
```diff
@@ -81,10 +81,10 @@
           </th>
         {% endif %}
         {% if not orderForViewing.delivered %}
-          <th class="text-right product_actions">
+          <th class="text-right product_actions d-print-none">
             <p>{{ 'Actions'|trans({}, 'Admin.Global') }}</p>
           </th>
-        {% endif %}
+          {% endif %}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig]:
```suggestion
        {% endif %}
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig [hunk 15]
```diff
@@ -32,55 +32,61 @@
...
-    <div class="col-md-4">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/customer.html.twig' %}
+    <div class="row">
+      <div class="col-md-4 left-column">
+        {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/customer.html.twig' %}
 
       {{ renderhook('displayBackOfficeOrderActions', {'id_order': orderForViewing.id}) }}
-
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/messages.html.twig' %}
+        {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/messages.html.twig' %}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig]:
```suggestion
      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/messages.html.twig' %}
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig [hunk 16]
```diff
@@ -32,55 +32,61 @@
...
+    </div>
 
-  <div class="row">
-    <div class="col-md-4">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/customer.html.twig' %}
+    <div class="row">
+      <div class="col-md-4 left-column">
+        {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/customer.html.twig' %}
 
       {{ renderhook('displayBackOfficeOrderActions', {'id_order': orderForViewing.id}) }}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig]:
```suggestion
        {{ renderhook('displayBackOfficeOrderActions', {'id_order': orderForViewing.id}) }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig [hunk 17]
```diff
@@ -32,55 +32,61 @@
...
-
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/messages.html.twig' %}
+        {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/messages.html.twig' %}
 
       {{ renderhook('displayAdminOrderSide', {'id_order': orderForViewing.id}) }}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig]:
```suggestion
        {{ renderhook('displayAdminOrderSide', {'id_order': orderForViewing.id}) }}
```

**Review by Progi1984 — COMMENTED** [review]
Some feedback

File: admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js (hunk 5 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js]: I don't now myself, that's from the original PR

File: admin-dev/themes/new-theme/scss/pages/_orders_view.scss (hunk 6 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/scss/pages/_orders_view.scss]: Well sorry, that's my past coming there, was working following BEM guidelines which is not bootstrap one, it should be only one dash for modifiers

File: admin-dev/themes/new-theme/template/components/layout/nav_bar.tpl (hunk 8 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/template/components/layout/nav_bar.tpl:25]: It should be, but if we decide that then everyone in the team should now adopt a complete component based guideline which permit every single component to be selected easily, even .row, .container, .col-md-* and stop using m*-* like classes but prefer creating a stylesheet for every components right named, else that would be a nonsense, some PR's would have this review, and others not

In that case that could be done CSS's side easily, I agree !

File: admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js (hunk 5 above)
Comment (matks) [reply admin-dev/themes/new-theme/js/pages/order/OrderViewPageMap.js]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: admin-dev/themes/new-theme/scss/pages/_orders_view.scss (hunk 6 above)
Comment (matks) [reply admin-dev/themes/new-theme/scss/pages/_orders_view.scss]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: admin-dev/themes/new-theme/scss/components/_button.scss (hunk 7 above)
Comment (matks) [reply admin-dev/themes/new-theme/scss/components/_button.scss]: Reverted in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig (hunk 9 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: admin-dev/themes/new-theme/scss/pages/_orders_view.scss (hunk 6 above)
Comment (matks) [reply admin-dev/themes/new-theme/scss/pages/_orders_view.scss]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig (hunk 10 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig (hunk 11 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_order_statistics.html.twig (hunk 12 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_order_statistics.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_title.html.twig (hunk 13 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/print_title.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig (hunk 14 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/products.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig (hunk 16 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig (hunk 17 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/880b0ffdaca18f032d2cb941f37e0751b4745479

**Review by atomiix — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig [hunk 18]
```diff
@@ -32,55 +32,61 @@
 
 
 {% block content %}
-  <div class="row">
-
-    {% set displayAdminOrderTopHookContent = renderhook('displayAdminOrderTop', {'id_order': orderForViewing.id}) %}
+  <div id="order-view-page">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig:35]: Please fix indent for this file

**Review by Progi1984 — COMMENTED** [review]
A last feedback

**Comment by matks** [comment]
> A last feedback

😭 

We definitely need a linter to tell us codestyle issues (indentation included) ...

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig (hunk 18 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig:35]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17617/commits/c5c493a09b80b93519121be4aeaa34e8f627d94e

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Many changes are not related to the pull request,  we can't find what really matter

**Review by Progi1984 — APPROVED** [review]
Approved, but don't forget that 2 approved are needed before QA

**Pull request closed without merging**